assets/search-form.js

```diff
@@ -3,15 +3,20 @@ class SearchForm extends HTMLElement {
     super();
     this.input = this.querySelector('input[type="search"]');
 
-    const form = this.querySelector('form.search');
-    form.addEventListener('reset', this.onFormReset.bind(this));
+    if (this.input) {
+      this.input.form.addEventListener('reset', this.onFormReset.bind(this));
+    }
+  }
+
+  shouldResetForm() {
+    return !document.querySelector('[aria-selected="true"] a')
   }
```

Comment:
I followed the pattern for the search button, that also does not submit when an element in the predictive search is active. Maybe @melissaperreault knows why this solution was adopted.
I don't know if it will be a lot of changes for one PR, but I could remove the `active` state and reset the list when the user tabs around, although that does not solve all the use cases because they could tab first and use the cursor later 🤔